### PR TITLE
Fix TestSolrCLIRunExample Nightly Thread Leak

### PR DESCRIPTION
**Pull request by risdenk**
TestSolrCLIRunExample has been failing thread leak in nightly since the Http2SolrClient has not been closed in `RunExampleTool`

### CODE REVIEW

**Review by HoustonPutman — CHANGES_REQUESTED** [review]
Looks good to me, but I don't think the inner try {} statements are needed, you can just catch the exception of the outer try and wrap in a SolrException, since that is done everytime.

File: solr/core/src/java/org/apache/solr/cli/RunExampleTool.java [hunk 1]
```diff
@@ -326,98 +326,98 @@ protected void runExample(CommandLine cli, String exampleName) throws Exception
             "exampledocs directory not found, skipping indexing step for the techproducts example");
       }
     } else if ("films".equals(exampleName) && !alreadyExists) {
-      SolrClient solrClient = new Http2SolrClient.Builder(solrUrl).build();
+      try (SolrClient solrClient = new Http2SolrClient.Builder(solrUrl).build()) {
+        echo("Adding dense vector field type to films schema \"_default\"");
+        try {
+          SolrCLI.postJsonToSolr(
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/cli/RunExampleTool.java:332]: Look at all the string handling for JSON!  I wondered about maybe using the FieldTypeDefinition.java, but after looking at how it's used, it seems to add another level of complexity/obfuscation over what is happening.   I hope our future V2 api's make doing thinks like adding a field type simpler and cleaner ;-).

**Review by epugh — APPROVED** [review]
LGTM.   THanks for fixing this!

**Comment by risdenk** [comment]
Thanks @epugh for the review and @HoustonPutman makes sense to make and for the commit.

**Pull request merged**